internal/controller/bootstrap/providerid_controller.go

```diff
@@ -88,7 +89,17 @@ func (p *ProviderIDController) Reconcile(ctx context.Context, req ctrl.Request)
 }
 
 func (p *ProviderIDController) SetupWithManager(mgr ctrl.Manager) error {
+	err := p.List(context.Background(), &clusterv1.MachineList{})
+	if err != nil && resourceNotFound(err) {
+		log.Log.Info("CAPI crds are not installed yet, skipping initializing providerID controller")
+		return nil
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.Machine{}).
 		Complete(p)
 }
+
+func resourceNotFound(err error) bool {
+	return strings.Contains(err.Error(), "the server could not find the requested resource")
```

Comment:
hmm, could we instead query the actual CRD via API resources instead of relying on somewhat arbitrary error messages? I think we do something like this in k0s side already in some cases.

Reply:
Yeah, makes sense. Changed it